### PR TITLE
Import automation: import progress dashboard frontend server

### PR DESCRIPTION
**Pull request by intrepiditee**
This PR adds the frontend server for the import progress dashboard. The server serves the webpage and retrieves resources from the REST API.

Thanks for reviewing.

### CODE REVIEW

File: import-automation/import-progress-dashboard-frontend/server/app/configs.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def get_dashboard_oauth_client_id():
+    """Gets the client ID used to authenticate with Identity-Aware Proxy
+    from the environment variable DASHBOARD_OAUTH_CLIENT_ID."""
+    return os.environ.get('DASHBOARD_OAUTH_CLIENT_ID')
+
+
+if _is_production():
+    _setup_logging()
```
Comment (jeffreyoldham) [inline import-automation/import-progress-dashboard-frontend/server/app/configs.py:43]: Bare code always worries me.

File: import-automation/import-progress-dashboard-frontend/server/app/main.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+@FLASK_APP.route('/<path:path>')
+def serve_file(path):
+    """Serves static files."""
+    return flask.send_from_directory(FLASK_APP.template_folder, path)
+
+
+@FLASK_APP.route('/system_runs')
+def get_system_runs():
+    """Returns all the system runs."""
+    # TODO(intrepiditee): Limit number returned
```
Comment (jeffreyoldham) [inline import-automation/import-progress-dashboard-frontend/server/app/main.py:55]:
```suggestion
    # TODO(intrepiditee): Limit number returned.
```

File: import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            google.auth.transport.requests.Request(), self.client_id)
+
+        # Fetch the Identity-Aware Proxy-protected URL, including an
+        # Authorization header containing "Bearer " followed by a
+        # Google-issued OpenID Connect token for the service account.
+        headers = kwargs.setdefault('headers', {})
+        headers['Authorization'] = f'Bearer {google_open_id_connect_token}'
+        response = requests.request(method, url, **kwargs)
+
+        return response
```
Comment (jeffreyoldham) [inline import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py:81]:
```suggestion
        return requests.request(method, url, **kwargs)
```

File: import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        # Obtain an OpenID Connect (OIDC) token from metadata server or
+        # using service account.
+        google_open_id_connect_token = id_token.fetch_id_token(
+            google.auth.transport.requests.Request(), self.client_id)
+
+        # Fetch the Identity-Aware Proxy-protected URL, including an
+        # Authorization header containing "Bearer " followed by a
+        # Google-issued OpenID Connect token for the service account.
+        headers = kwargs.setdefault('headers', {})
+        headers['Authorization'] = f'Bearer {google_open_id_connect_token}'
```
Comment (jeffreyoldham) [inline import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py:78]: So "headers" is a pointer apparently.

File: import-automation/import-progress-dashboard-frontend/server/app/main.py (hunk 2 above)
Comment (intrepiditee) [reply import-automation/import-progress-dashboard-frontend/server/app/main.py:55]: fixed

File: import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py (hunk 3 above)
Comment (intrepiditee) [reply import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py:81]: fixed

File: import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py (hunk 4 above)
Comment (intrepiditee) [reply import-automation/import-progress-dashboard-frontend/server/app/service/iap_request.py:78]: What pointer?

File: import-automation/import-progress-dashboard-frontend/server/app/configs.py (hunk 1 above)
Comment (intrepiditee) [reply import-automation/import-progress-dashboard-frontend/server/app/configs.py:43]: moved to create_app

**Pull request merged**